### ISSUE TITLE
Customer get unsubscribe to newsletter on password reset email request with Newsletter Need to Confirm Set to Yes on admin settings

### ISSUE THREAD

**Issue by matin73** [magento/magento2#17954]
### Preconditions

1. Magento 2.2.4 and 2.2.5
2. PHP 7.0
3.Open Admin Stores > Configuration > Customers > Newsletter > Need to Confirm Set to Yes and save.

### Steps to reproduce

1. Go to Storefront.
2. Create customer account and "Sign in".
3. Go to "Newsletter Subscription" in account page.
4. Activate checkbox  "General Subscription" , a confirmation request has been sent on email with link, dont click on this. 
5. Sign out of customer and go to link "Sign In" click "Forgot Your Password" and fill field to reset password and press button "Reset My Password"
6. Check you email

### Expected result

1. Customer should stay "Subscribes to newsletter".
2. Letter with "Set new password" button is present on email.

### Actual result

1. Letter "You have been unsubscribed from the newsletter" on email.
2. Letter with "Set new password" button is present on email.

**Comment by magento-engcom-team** [comment]
Hi @matin73. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@matin73 do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by matin73** [comment]
@matin73 do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?

 x   yes
    no

**Comment by ghost** [comment]
@matin73 ,  thank you for your report.
We were not able to reproduce this issue by following the steps you provided. Please provide more detailed steps to reproduce or try to reproduce this issue on a clean installation or latest release.

Only 3 letter on email box is present and customer is not unsubscription.
See scrennshots

![letter_on_email](https://user-images.githubusercontent.com/41946904/45161630-fa0b9600-b1f4-11e8-86e8-4201b756c673.png)

![register](https://user-images.githubusercontent.com/41946904/45161873-6be3df80-b1f5-11e8-88df-cd51d2ec5283.png)

![sub](https://user-images.githubusercontent.com/41946904/45161900-7605de00-b1f5-11e8-85c3-6a32455bba87.png)

![reset_password](https://user-images.githubusercontent.com/41946904/45161910-7d2cec00-b1f5-11e8-8a9b-d22cf0a6b177.png)

**Comment by matin73** [comment]
maybe it depends on some configuration. here is my setup
![grafik](https://user-images.githubusercontent.com/3177275/45163913-a6e41400-b1f1-11e8-8110-9a8a68f4dcae.png)


![grafik](https://user-images.githubusercontent.com/3177275/45163870-903dbd00-b1f1-11e8-98f6-c8e6a18927ac.png)

![grafik](https://user-images.githubusercontent.com/3177275/45163861-8d42cc80-b1f1-11e8-8c91-77861ab6f6bc.png)

**Comment by matin73** [comment]
And maybe you have to make at least 1 order (at both systems customer made at least 1 order)

Edit: I created new customer without order - same problem.

**Comment by matin73** [comment]
I think unsubscribe email is sent, if customer has not confirmed newsletter subscription yet ...!?

**Comment by matin73** [comment]
I have testet in now once again. Steps to reproduce:
1) Intall Magento 
2) Create customer account. Sign in and go to "Newsletter Subscription" in account.
3) Activate checkbox for newsletter subscription " General Subscription"
4) Now email "Newsletter subscription confirmation" is sent. Now do not click activation link in email.
5) Log out from account and click "Forgot you passwort" link and enter email
6) And now email "Newsletter unsubscription success" is automatically sent

The next thing ist following: This is very annoying because we importet 13.000 customer from old shop system. And all customer have to reset password because password could not be importet. So every  customer gets FIRST another email that he has been SUBSCRIBED to newsletter and next email is that customer is unsubscribed. This is confusing customer becauser they do not want to get any newsletter!

**Comment by ghost** [comment]
@matin73 , thanks for you audate. 
We've acknowledged the issue and added to our backlog.

**Comment by matin73** [comment]
How can I fix this? I do not need newsletter at all. I tried to disable module by:
php bin/magento module:disable --clear-static-content Magento_Newsletter
php bin/magento setup:upgrade

But I get error:
Unable to change status of modules because of the following constraints:
Cannot disable Magento_Newsletter because modules depend on it:
Magento_Store: Magento_Store->Magento_Catalog->Magento_Customer->Magento_Newsletter
Magento_Directory: Magento_Directory->Magento_Backend->Magento_Customer->Magento_Newsletter
Magento_Eav: Magento_Eav->Magento_Backend->Magento_Customer->Magento_Newsletter
Customweb_Base: Customweb_Base->Magento_Customer->Magento_Newsletter
Magento_Theme: Magento_Theme->Magento_Customer->Magento_Newsletter
Magento_Backend: Magento_Backend->Magento_Customer->Magento_Newsletter
Magento_Customer: Magento_Customer->Magento_Newsletter

**Comment by rparsi** [comment]
@matin73 You can prevent the sending of newsletter emails by overriding (DI mechanism)
class `Magento\Newsletter\Model\Subscriber`, sample code below:

```
namespace MyVendor\Newsletter\Model;

use Magento\Newsletter\Model\Subscriber as Base;

class Subscriber extends Base
{
    public function sendConfirmationRequestEmail()
    {
        // do nothing here so that base class doesn't send emails
        return $this;
    }

    public function sendConfirmationSuccessEmail()
    {
        // do nothing here so that base class doesn't send emails
        return $this;
    }

    public function sendUnsubscriptionEmail()
    {
        // do nothing here so that base class doesn't send emails
        return $this;
    }
}
```

Make sure your `app/code/MyVendor/Newsletter/etc/di.xml` has
```
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
    <preference for="Magento\Newsletter\Model\Subscriber" type="MyVendor\Newsletter\Model\Subscriber"/>
</config>
```

**Comment by matin73** [comment]
@rparsi thanks. but to complicated

**Comment by rparsi** [comment]
@matin73 How is the solution complicated?  I gave you all the necessary code, just change `MyVendor` to something appropriate for your application.
Your only other option is to find an off the shelf 3rd party extension that manages transactional emails.

**Comment by magento-engcom-team** [comment]
Hi @janakbhimani. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by sidolov** [comment]
Hi @matin73. Thank you for your report. 
The issue has been fixed in magento/magento2#18643 by @janakbhimani in 2.2-develop branch
Related commit(s):
 - [332f88f209db78e3194cb6b0caa35310309419da](https://github.com/magento/magento2/commit/332f88f209db78e3194cb6b0caa35310309419da)
 - [8c868c3c315e15a40671ee72c24e638172c8d5a6](https://github.com/magento/magento2/commit/8c868c3c315e15a40671ee72c24e638172c8d5a6)
 - [26e4decae625b1df1b67cb5c29f2535d837fc214](https://github.com/magento/magento2/commit/26e4decae625b1df1b67cb5c29f2535d837fc214)

The fix will be available with the upcoming 2.2.8 release.

**Comment by sidolov** [comment]
Hi @matin73. Thank you for your report. 
The issue has been fixed in magento/magento2#18795 by @janakbhimani in 2.3-develop branch
Related commit(s):
 - [6e9d5e5fcda4ce91ef25559bb946c8e52e777148](https://github.com/magento/magento2/commit/6e9d5e5fcda4ce91ef25559bb946c8e52e777148)

The fix will be available with the upcoming 2.3.1 release.
